src/main.rs

```diff
@@ -26,7 +30,17 @@ async fn main() -> anyhow::Result<()> {
     let args = Args::parse();
 
     let client = Client::try_default().await?;
-    let pod_list = args.pods;
+
+    let mut pod_list = args.pods;
+
+    if args.deployments.len() > 0 {
+        for deploy in args.deployments.iter() {
+            pod_list.append(
+                &mut k8s::get_pod_list_for_deployment(&client, &deploy, &args.namespace).await?,
```

Comment:
🚫 **[clippy]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> 
<pre><code>error: this expression creates a reference which is immediately dereferenced by the compiler
  --> src/main.rs:39:64
   |
39 |                 &mut k8s::get_pod_list_for_deployment(&client, &deploy, &args.namespace).await?,
   |                                                                ^^^^^^^ help: change this to: `deploy`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
   = note: `-D clippy::needless-borrow` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::needless_borrow)]`
</code></pre>
<!-- __reviewdog__:ChBiYjM1NDhmYmY1NmFlN2E1EgZjbGlwcHk= -->